CWE.Data/DataAccessLayer.cs

```diff
@@ -132,5 +134,122 @@ public async Task<IEnumerable<Request>> GetRequests()
             return await this.dbContext.Requests
                 .ToListAsync();
         }
+
+        /// <summary>
+        /// Checks if the tag name provided exists in the database.
+        /// </summary>
+        /// <param name="tagName">The name of the tag to fetch..</param>
+        /// <returns>A <see cref="Tag"/> depending on if the tag exists in the database.</returns>
+        public async Task<Tag> FetchTagAsync(string tagName)
```

Comment:
```suggestion
        public async Task<Tag> GetTag(string tagName)
```
Same naming as previous methods